numpy/_core/src/multiarray/methods.c

```diff
@@ -2756,6 +2756,32 @@ array_class_getitem(PyObject *cls, PyObject *args)
     return Py_GenericAlias(cls, args);
 }
 
+static PyObject *
+array_array_namespace(PyArrayObject *self, PyObject *args, PyObject *kwds)
+{
+    static char *kwlist[] = {"api_version", NULL};
+    char *array_api_version = "2022.12";
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "|$s:__array_namespace__", kwlist,
+                                     &array_api_version)) {
+        return NULL;
+    }
+
+    if (strcmp(array_api_version, "2022.12") != 0) {
```

Comment:
Calling `ndarray.__array_namespace__(api_version='2021.12')` should work too. The newer version is a superset of the older one, so we'd want to support both. And when we upgrade to say `2023.12` in the future, asking for `2022.12` should not break. So adding a test to check this will be useful.

Reply:
Second suggested regression test:
```python
>>> import numpy as np
>>> xp = np.arange(2).__array_namespace__()
>>> assert xp is np
```

Reply:
I added `test__array_namespace__` for that purpose.